crates/executor/src/executor.rs

```diff
@@ -397,33 +403,149 @@ where
                     block_available_gas,
                 })
             }
-            // Execute transaction.
-            let ResultAndState { result, state } = self.transact(transaction, sender)?;
-
-            // commit changes
-            self.commit_changes(
-                state,
-                self.chain_spec.fork(Hardfork::SpuriousDragon).active_at_block(block.number),
-                &mut post_state,
-            );
 
-            // append gas used
-            cumulative_gas_used += result.gas_used();
-
-            // cast revm logs to reth logs
-            let logs: Vec<Log> = result.logs().into_iter().map(into_reth_log).collect();
-
-            // Push transaction changeset and calculate header bloom filter for receipt.
-            post_state.add_receipt(Receipt {
-                tx_type: transaction.tx_type(),
-                // Success flag was added in `EIP-658: Embedding transaction status code in
-                // receipts`.
-                success: result.is_success(),
-                cumulative_gas_used,
-                bloom: logs_bloom(logs.iter()),
-                logs,
-            });
-            post_state.finish_transition();
+            #[cfg(feature = "optimism")]
+            {
+                let db = self.db();
+                let l1_cost = l1_cost_oracle
+                    .calculate_l1_cost(db, block.header.number, transaction)
+                    .map_err(|db_err| Error::DBError { inner: db_err.to_string() })?;
+
+                let sender_account = db.load_account(sender).map_err(|_| Error::ProviderError)?;
+                let old_sender_info = to_reth_acc(&sender_account.info);
+                if let Some(m) = transaction.mint() {
+                    // Add balance to the caller account equal to the minted amount.
+                    // Note: this is unconditional, and will not be reverted if the tx fails
+                    // (unless the block can't be built at all due to gas limit constraints)
+                    sender_account.info.balance += U256::from(m);
+                }
```

Comment:
This could have been addressed in Revm's transact() function in a more clean way. In fact I have a local branch of what this change would look like here: https://github.com/merklefruit/revm/pull/1/files#diff-1d478ba44ccc56e3b1142bd3723bf97f3e254c25dd18323481aedadce0803e91R130

However that solution has some cons:
1. we are not able to revert the deposit if the block runs out of gas, because revm's transact() context doesn't know about the outstanding block gas used (called `cumulative_gas_used` here)
2. we can't cache the values to calculate `l1_cost` for the entire block, and would have to access the DB for each transaction in the block instead. This is the main reason why I wanted to keep all the diffs on Op-reth if possible